### ISSUE TITLE
[Feature Request] data source bitwarden_item_login without id but selecting by name, website or other fields

### ISSUE THREAD

**Issue by bb1** [maxlaverse/terraform-provider-bitwarden#123]
Hi there,

it would be great if the bitwarden_item_login would not require an id but could also select resources by other means. Alternatively this could be archived by a new data source maybe bitwarden_item_list ?

### Affected Resource(s)
Please list the resources as a list, for example:
- bitwarden_item_login

### Terraform Configuration Files
```hcl
# old
data "bitwarden_item_login" "database_credentials" {
  id = "e88682d6-197c-4d7f-abd5-35e0e11db6fb"
}

# new
data "bitwarden_item_login" "database_credentials" {
  name = "My admin dashboard login"
}

data "bitwarden_item_login" "database_credentials" {
  website = "https://admin.secret-url.de"
  organizationid = "e88682d6-197c-4d7f-abd5-35e0e11db6fb"
}
```

### References
This could be done by using the list funciton of the bw cli.
- https://bitwarden.com/help/cli/#list
or the vault REST API through the bw cli:
- https://bitwarden.com/help/vault-management-api/#operations-tag-Vault_Items /list/object/items

### Off-Topic
Great project. I would love to add this myself but I never did go. Anyway keep up the good work!

**Comment by maxlaverse** [comment]
Hi @bb1 ! Sounds like a good idea. I'll do some research and try to get this feature out.

**Comment by timvy** [comment]
This would be a great option!

**Comment by maxlaverse** [comment]
Hi @bb1,
I ended up implementing search filters on data sources staying close to the CLI's interface. Instead of having to provide an `id`, you can now provide a `search` attribute and eventually filter out the results by additional parameters (e.g. organizationid).

``` terraform
data "bitwarden_item_login" "example" {
  search = "Example"
}
```

It's released in [v0.8.0](https://github.com/maxlaverse/terraform-provider-bitwarden/releases/tag/v0.8.0)
